.github/workflows/appimage.yml

```diff
@@ -0,0 +1,124 @@
+# Manually triggered build of mlnet, mlgui and utils binaries and appimage with mlnet+gui
+
+name: Build AppImage
+
+on:
+  workflow_dispatch
+
+env:
+  APPIMAGEBUILDER_VERSION: 1.1.0
+  APPIMAGEBUILDER_SHA1SUM: 6f83a789f6c47a745b97d1b0b3f1df2e7eea7a09
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-22.04]
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Install dependencies
+        run:  sudo apt-get install -y --no-install-recommends autoconf ocaml ocaml-findlib liblablgtk2-ocaml-dev camlp4 libnum-ocaml-dev libminiupnpc-dev libnatpmp-dev libbz2-dev librsvg2-dev libmagic-dev libgtk2.0-dev liblablgtk2-ocaml-dev liblablgtk2-gl-ocaml-dev liblablgtk2-gnome-ocaml-dev libgd-dev
+
+      - name: Configure options
+        run:  ./configure --enable-batch --enable-upnp-natpmp --disable-directconnect --disable-fasttrack --disable-gnutella --disable-gnutella2 --enable-gui=newgui2
+
+      - name: Build binaries
+        run:  make
+
+      - name: Build utils
+        run:  make utils
+
+      - name: Save version
+        run:  |
+              VERSION=$(./mlnet -v 2>&1 | sed -ne "1s#^.*Starting MLDonkey \([^ ]\+\) .*#\1#p")
```

Comment:
there should be better way to get version without running the binary. Worst case can grep configure log, or actually this : 
```
$ ocaml -e '#use "src/utils/lib/autoconf.ml";; print_endline current_version'
3.2.1.git
```
see https://github.com/ocaml/ocaml/issues/13637 for why less ugly methods don't work